include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -142,6 +152,117 @@ class alignment_algorithm :
         return align_result{res};
     }
 
+    /*!\brief Invokes the banded alignment computation given two sequences.
+     * \tparam    first_batch_t  The type of the first sequence (or packed sequences); must model std::ForwardRange.
+     * \tparam    second_batch_t The type of the second sequence (or packed sequences); must model std::ForwardRange.
+     * \param[in] first_batch    The first sequence (or packed sequences).
+     * \param[in] second_batch   The second sequence (or packed sequences).
+     *
+     * \details
+     *
+     * Computes the k-banded alignment. This function is only available if the alignment configuration was configured
+     * with seqan3::align_cfg::band.
+     *
+     * ### Exception
+     *
+     * Strong exception guarantee. Might throw std::bad_alloc or seqan3::invalid_alignment_configuration.
+     *
+     * ### Thread-safety
+     *
+     * Calls to this functions in a concurrent environment are not thread safe. Instead use a copy of the alignment
+     * algorithm type.
+     *
+     * ### Complexity
+     *
+     * The code always runs in \f$ O(N*k) \f$ time and depending on the configuration requires at least \f$ O(k) \f$
+     * and at most \f$ O(N*k) \f$ space.
+     */
+    template <std::ranges::ForwardRange first_batch_t, std::ranges::ForwardRange second_batch_t>
+    auto operator()(first_batch_t const & first_batch, second_batch_t const & second_batch)
+    //!\cond
+        requires is_banded
+    //!\endcond
+    {
+        assert(cfg_ptr != nullptr);
+
+        using std::get;
+
+        static_assert(config_t::template exists<align_cfg::band>(),
+                      "The band configuration is required for the banded alignment algorithm.");
+        auto const & band = get<align_cfg::band>(*cfg_ptr).value;
+
+        // ----------------------------------------------------------------------------
+        // Check valid band settings.
+        // ----------------------------------------------------------------------------
+
+        using diff_type = typename std::iterator_traits<std::ranges::iterator_t<first_batch_t>>::difference_type;
+        static_assert(std::is_signed_v<diff_type>,  "Only signed types can be used to test the band parameters.");
+
+        if (static_cast<diff_type>(band.lower_bound) >
+            std::distance(seqan3::begin(first_batch), seqan3::end(first_batch)))
```

Comment:
Negative bands are allowed as long as they intersect with the matrix. 
batch is a either a single range or a range of simd vector

Reply:
> Negative bands are allowed as long as they intersect with the matrix.

But a negative number will never be bigger than a distance which is >= zero  :thinking: 

> batch is a always a single range over either a single alphabet or a simd vector

And this line covers both? Wouldn't you need to add a factor for the case of collection-of-simd-vectors?

Reply:
upper and lower bound can be both negative, positive or negative and positive.
This line says if the value for the lower bound is bigger than the length of the sequence in the horizontal direction, the band is invalid. The simd sequence is as long as the longest sequence.
So for some sequences it could be invalid. But we deal with this when we need to. The other case (upper diagonal exceeds the length of the vertical sequence) is handled below.

Reply:
> upper and lower bound can be both negative, positive or negative and positive.
> This line says if the value for the lower bound is bigger than the length of the sequence in the horizontal direction, the band is invalid.

Got this!

> The simd sequence is as long as the longest sequence.

I still don't understand how this function can handle batches of sequences, the distance will just return the size of the batch and not the size of the sequences :question:

Reply:
Well, it depends on the behaviour we want. All smaller sequences will be padded with some extra symbols so we can compute the alignment and later it will be resolved to practically aligning seq1 to the empty seq and vice versa for seq2. Or we refactor it later to check the individual sequences and throw. But this comes anyway later.

Reply:
can we talk about this on gitter or on the phone? I have the feeling I am missing something o_O

Reply:
Ok, I got it now. 

As discussed in the chat I would hugely appreciate a rename to something normal like "rng_t" oder "sequence_t" with a description in the documentation that states something like `In the case of generic inter-sequence vectorisation this is a range over SIMD vectors instead of a range over alphabet letters.`
The regular ranges would be over the original seqan3::Alphabet but in SIMD case they would be over simd vectors of the ranks, right?

Reply:
It's fine for me and yes the last statement is right